src/preferences/game.cpp

```diff
@@ -380,28 +380,43 @@ void _set_lobby_joins(int show)
 	}
 }
 
-const std::vector<game_config::server_info>& server_list()
-{
-	static std::vector<game_config::server_info> pref_servers;
-	if(pref_servers.empty()) {
-		std::vector<game_config::server_info> &game_servers = game_config::server_list;
-		VALIDATE(!game_servers.empty(), _("No server has been defined."));
-		pref_servers.insert(pref_servers.begin(), game_servers.begin(), game_servers.end());
-		for(const config &server : get_prefs()->child_range("server")) {
-			game_config::server_info sinf;
-			sinf.name = server["name"].str();
-			sinf.address = server["address"].str();
-			pref_servers.push_back(sinf);
-		}
+const std::vector<game_config::server_info>& builtin_servers_list()
+{
+	static std::vector<game_config::server_info> pref_servers = game_config::server_list;
+	return pref_servers;
+}
+
+std::vector<game_config::server_info> user_servers_list()
+{
+	std::vector<game_config::server_info> pref_servers;
+
+	for(const config &server : get_prefs()->child_range("server")) {
+		game_config::server_info sinf;
+		sinf.name = server["name"].str();
+		sinf.address = server["address"].str();
+		pref_servers.push_back(sinf);
```

Comment:
You can use `emplace_back` here.

Reply:
There's a (redundant) default constructor explicitly defined for `game_config::server_info` which interferes with this and other things such as move assignment.